recipe/2025-08-recipe-packs.md

```diff
@@ -0,0 +1,379 @@
+
+## Terms and definitions
+
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Recipe | IaC templates that operators register on a Radius Environment |
+| Recipe Pack| A collection of recipes that can be managed as an entity |
+| RRT | [Radius Resource Type](https://docs.radapp.io/guides/author-apps/custom/overview/)|
+
+## Objectives
+
+> **Issue Reference:** 
+
+### Goals
+
+Provide Recipe Packs as a Radius feature to bundle multiple recipes as a single managable entity into a Radius Environment (e.g., a pack for ACI that includes all necessary recipes).
+
+### Non goals
+
+Recipe Packs would bundle together Recipes, as we understand them today. We do not cover recipe versioning / other recipe specific enhancements in this design. 
+
+
+### User scenarios (optional)
+
+#### Registering several recipes to an environment
+
+As an operator I am responsible for creating Radius Environments using which developers can deploy their applications. As part of creating the environment, I manually link recipes one by one using `rad recipe register` or by updating the environment definition. This can be error prone when there are many recipes and environments.  Radius should provide a way to bulk register( and manage) recipes.
+
+#### Registering recipes to multiple environment
+
+As an operator I am responsible for creating Radius Environments using which developers can deploy their applications. As part of creating the environment, I manually link recipes one by one using `rad recipe register` or by updating the environment definition. I having 100s of environment which mostly use the same recipes. Piecing the same recipes together for each environment feels like rework.
+
+## Design
+
+### Design Overview
+
+We choose modelling Recipe Pack as a first class Radius Resource Type. [Alternatives considered](#alternatives-considered) section details some other possibilities.
+
+Pros:
+
+- Solves the requirement for bulk registering recipes using single command with a one time effort of creating the recipe pack resource
+- As first class resource, recipe packs would be displayed in app graphs. They can also have their own lifecycle and rbac independant of environments. 
+- Helps reduce the size of environment resource, which could reach serization limits with tons of recipes. 
+- Helps reduce overall size of Radius datastore, since common recipe information could now be stored as a single resource instead of being duplicated across several environments.
+
+[Question]: would environment and recipe pack ever have different rbac?
```

Comment:
I don't think Recipe Packs have RBAC? They should just inherit whatever RBAC is needed to drive its constituent Recipes, which in turn inherits RBAC from the Environment it's registered to?